syntax = "proto3";

option go_package = ".;pb";

service Face {
    rpc Comparison(stream ComparisonRequest) returns (ComparisonRespons){}
    rpc Find(stream FindRequest) returns (FindRespons) {}
}

message ComparisonRequest {
    oneof ComparisonData {
        Metadata originalMetadata = 1;
        bytes originalImage = 2;
        Metadata forCheck = 3;
        bytes forCheckImage = 4;
    }
}
message ComparisonRespons {
    bool coincidences = 1;
}

message FindRequest {
    oneof FindData {
        Metadata metadata = 1;
        bytes image = 2;
    }
}
message FindRespons {
    int32 total = 1;
}

message Metadata{
    string filename = 1;
    string extension = 2;
}